src/lib/y2storage/disk_size.rb

```diff
@@ -609,10 +609,48 @@ def floor(unit_size)
     # @example
     #   x = DiskSize.KB(1)   #=> <DiskSize 0.98 KiB (1000)>
     #   x.to_human_string    #=> "0.98 KiB"
-    def to_human_string
+    def to_human_string(round_method = :round)
       return "unlimited" if unlimited?
-      size, unit = human_string_components
-      format("%.2f %s", size, unit)
+      float, unit_s = human_string_components
+      rounded = (float * 100).public_send(round_method) / 100.0
+      # A plain "#{rounded} #{unit_s}" would not keep trailing zeros
+      format("%.2f %s", rounded, unit_s)
+    end
+
+    # A human readable string that does not exceed the exact size.
+    #
+    # If we have 4.999 GiB of space, and ask the user how much of that
+    # should be used, prefilling the "Size" widget with the maximum
+    # rounded up to "5.00 GiB" it will then fail validation.
```

Comment:
The use case is the Partition Size dialog, where you can choose
- Use the maximum size available (4.99GiB)
- Use a specific size: `4.99GiB`
- Use a specific region, from `12345` to `19999`

![2283](https://user-images.githubusercontent.com/102056/27337771-b2fbe5e0-55d3-11e7-8ad3-ad75fb721acc.png)

and yes, if you select the specific `4.99GiB` size, you will waste up to 1% of space. I don't know of a way to avoid this without confusing the user with subtle details.

Oh well, we could say: "Use a specific size: `5368601746` " which is **exact**, but slightly less than 5GiB. Would you know the difference?

Regarding "validation", is this better?
> then fail validation (checking that the entered value fits in the available space)